components/action-mgt/org.wso2.carbon.identity.action.execution/src/main/java/org/wso2/carbon/identity/action/execution/ActionExecutorServiceImpl.java

```diff
@@ -315,13 +324,12 @@ private ActionExecutionStatus processSuccessResponse(Action action,
                                                                  actionExecutionResponseProcessor)
             throws ActionExecutionResponseProcessorException {
 
-        if (LOG.isDebugEnabled()) {
-            // todo: add to diagnostic logs
+        if (LOG.isDebugEnabled() || LoggerUtils.isDiagnosticLogsEnabled()) {
```

Comment:
we can move this if condition within logSuccessResponse and with that you can avoid one more if condition as your logger handles it

Reply:
Updated in